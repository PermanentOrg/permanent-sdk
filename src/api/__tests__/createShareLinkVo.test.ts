import nock from 'nock';
import { createShareLinkVo } from '..';

describe('createShareLink', () => {
  test('should create a share link', async () => {
    nock('https://permanent.local')
      .post(
        '/api/share/generateShareLink',
        {
          folderLinkId: 8306,
          maxUses: 0,
          previewToggle: false,
          defaultAccessRole: 'access.role.viewer',
        },
      )
      .replyWithFile(
        200,
        `${__dirname}/fixtures/createShareLink/shareLinkVo.json`,
        { 'Content-Type': 'application/json' },
      );

    const shareLinkVo = await createShareLinkVo(
      {
        bearerToken: '12345',
        baseUrl: 'https://permanent.local/api',
      },
      {
        folderLinkId: 8306,
        maxUses: 0,
        previewToggle: false,
        defaultAccessRole: 'access.role.viewer',
      },
    );

    expect(shareLinkVo).toMatchSnapshot();
  });
});
