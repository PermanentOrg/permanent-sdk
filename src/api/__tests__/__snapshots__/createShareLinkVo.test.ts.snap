// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createShareLink should create a share link 1`] = `
{
  "autoApproveToggle": null,
  "byAccountId": 1770,
  "byArchiveId": 1765,
  "createdDT": "2024-01-31T23:41:54",
  "defaultAccessRole": "access.role.viewer",
  "expiresDT": null,
  "folder_linkId": 8306,
  "maxUses": 0,
  "previewToggle": 0,
  "shareUrl": "https://local.permanent.org/share/b6b457c4c869d05bb8977ea4be6e66d3957442ce7dcc5892354faf084557a805",
  "shareby_urlId": 173,
  "status": "status.generic.ok",
  "updatedDT": "2024-01-31T23:41:54",
  "urlToken": "b6b457c4c869d05bb8977ea4be6e66d3957442ce7dcc5892354faf084557a805",
  "uses": 0,
}
`;
